extensions/interactions/InteractiveMap/directives/InteractiveMap.js

```diff
@@ -38,13 +41,23 @@ oppia.directive('oppiaInteractiveInteractiveMap', [
             HtmlEscaperService.escapedJsonToObj($attrs.longitudeWithValue)];
           $scope.zoom = (
             HtmlEscaperService.escapedJsonToObj($attrs.zoomWithValue));
+          $scope.interactionIsActive = ($scope.getLastAnswer() === null);
+          $scope.mapMarkers = [];
+
+          $scope.$on(EVENT_NEW_CARD_AVAILABLE, function() {
+            $scope.interactionIsActive = false;
+            $scope.overlayStyle = {
```

Comment:
There's overlap in functionality between this and lines 93-98. We should pull that out into a separate function.

It might be better to have separate functions for setOverlay() and hideOverlay(), and name the existing functions onMouseover() and onMouseout().